### DISCUSSION TITLE
Detection Training - RuntimeWarning divide by zero

### DISCUSSION THREAD

**Discussion by ArsalanYounus007** [mindee/doctr#1315] [General]
Hello,

I am trying to train db_resnet50 for Checkbox (detection). I created my own annotations and started the training.

I am getting 
`F:\GitHub\doctr\doctr\models\detection\differentiable_binarization\base.py:247: RuntimeWarning: divide by zero encountered in true_divide`

Loss goes up whenever this occurs, I am sure that this is a problem with my dataset annotations, But I am unable to figure out what.

**Comment by ArsalanYounus007** [comment]
Is it because the GTs are too close to each other?? I have used the show-sample flag to get this image

![Figure_1](https://github.com/mindee/doctr/assets/34642875/25bdb997-e9e7-4f1e-a413-5dc27c61798c)

**Reply by felixT2K** [reply]
Hi @ArsalanYounus007 👋🏼,

The reason why this happens is, that we filter out to small boxes which can't be a character like noise in an image which is detected as character (possible dot).
One option would be to play a bit around with the following both values to expand the seq map detections:
https://github.com/mindee/doctr/blob/7ab0ece3a4aec256bb36c311f4c6eeaa597d826b/doctr/models/detection/differentiable_binarization/base.py#L35

**Reply by ArsalanYounus007** [reply]
Hi, Thank you for your reply.
I have changed the value to very small for example: 0.001 - 0.01 but I still see the warning and loss jumps on each sample.

**Comment by ArsalanYounus007** [comment]
One more thing, Even though I let it train with Warnings. I got a checkpoint - I know it might not be a great checkpoint but I thought to load and test it anyways

When I load it (using: https://mindee.github.io/doctr/using_doctr/custom_models_training.html) , I get this error. 
![image](https://github.com/mindee/doctr/assets/34642875/fe541b7c-d2fc-4014-86f0-dde0d5c5d2b9)

**Reply by felixT2K** [reply]
See: https://mindee.github.io/doctr/using_doctr/custom_models_training.html   :)

**Reply by ArsalanYounus007** [reply]
Eish... My bad!!

**Reply by ArsalanYounus007** [reply]
![image](https://github.com/mindee/doctr/assets/34642875/02276e69-c93e-465e-822e-3e1d00430056)

As expected, Got nothing on inference
Back to my original question. How do I fix the training of Checkboxes?

**Reply by felixT2K** [reply]
Hi @ArsalanYounus007 
You can try to modify the shrink value and min_size_box by lowering both values https://github.com/mindee/doctr/blob/7ab0ece3a4aec256bb36c311f4c6eeaa597d826b/doctr/models/detection/differentiable_binarization/base.py#L165 or expand your box coordinates in your dataset a bit.
How many samples are in your dataset ?

**Reply by ArsalanYounus007** [reply]
I have around 1500 pages in total with CheckBoxes. 
I will play with these parameters and see if it works, Thank you!!
